include/ensmallen_bits/callbacks/early_stop_at_min_loss.hpp

```diff
@@ -28,11 +33,33 @@ class EarlyStopAtMinLoss
    * @param patienceIn The number of epochs to wait after the minimum loss has
    *    been reached or no improvement has been made (Default: 10).
    */
-  EarlyStopAtMinLoss(const size_t patienceIn = 10) :
-      patience(patienceIn),
+  EarlyStopAtMinLoss<MatType>(
+      const size_t patienceIn = 10,
+      std::ostream& output = arma::get_cout_stream())
```

Comment:
I'm not sure about the use of an `ostream` here.  This means that `EarlyStopAtMinLoss` behaves in two really different ways: (1) don't pass in a callback, and nothing is printed; or (2) pass in a callback, that is used to compute the objective, and its value is printed all the time.  I think that maybe we should remove `output` from here, and perhaps later consider some way that we can print the validation loss in, e.g., `ProgressBar` or something like that.

Reply:
Exactly, this is what is happening, If we do not passe lambda nothing will be printed.
It is true that this is confusing. I can print it locally from inside the lambda, in this case, there is no need to have `output` in this class.